File: android/301Project/app/src/main/java/com/example/a301project/shopping/adapter/ShoppingListAdapter.java
package com.example.a301project.shopping.adapter;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.example.a301project.R;
import com.example.a301project.base.BaseAdapter;
import com.example.a301project.shopping.entity.ShoppingItem;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


public class ShoppingListAdapter extends BaseAdapter {
    public ShoppingListAdapter(List<?> dataList, Context context) {
        super(dataList, context);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = layoutInflater.inflate(R.layout.shopping_item, viewGroup, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int position) {
        ViewHolder holder = (ViewHolder) viewHolder;
        ShoppingItem shoppingItem = (ShoppingItem) dataList.get(position);
        holder.tv_shopping_item_ingredient.setText(shoppingItem.getIngredient());
        holder.tv_shopping_item_amount.setText(shoppingItem.getAmount());
        holder.cb_shopping_item.setChecked( shoppingItem.isSelected());

        holder.cb_shopping_item.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                shoppingItem.setSelected(b);
            }
        });
        holder.tv_shopping_item_unit.setText(shoppingItem.getUnit());
        holder.tv_shopping_item_category.setText(shoppingItem.getCategory());
        holder.ll_shopping_item.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onItemClickListener.onClickListener(view,position);
            }
        });
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView tv_shopping_item_ingredient;
        TextView tv_shopping_item_amount;
        CheckBox cb_shopping_item;
        TextView tv_shopping_item_unit;
        TextView tv_shopping_item_category;
        LinearLayout ll_shopping_item;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tv_shopping_item_ingredient = itemView.findViewById(R.id.tv_shopping_item_ingredient);
            tv_shopping_item_amount = itemView.findViewById(R.id.tv_shopping_item_amount);
            cb_shopping_item = itemView.findViewById(R.id.cb_shopping_item);
            tv_shopping_item_unit = itemView.findViewById(R.id.tv_shopping_item_unit);
            tv_shopping_item_category = itemView.findViewById(R.id.tv_shopping_item_category);
            ll_shopping_item = itemView.findViewById(R.id.ll_shopping_item);
        }




    }
}

File: android/301Project/app/src/main/java/com/example/a301project/shopping/adapter/ShoppingListAdapter.kt
package com.example.a301project.shopping.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.recyclerview.widget.RecyclerView
import com.example.a301project.R
import com.example.a301project.base.BaseAdapter
import com.example.a301project.shopping.entity.ShoppingItem

/**
 * @ClassName: DownloadListAdapter
 * @Description: java类作用描述
 * @Author: ping
 * @CreateDate: 2022/9/5 5:55 下午
 */
class ShoppingListAdapter(dataList: MutableList<ShoppingItem>, context: Context) : BaseAdapter(dataList, context) {



    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): RecyclerView.ViewHolder {
        val view: View = layoutInflater.inflate(R.layout.shopping_item, viewGroup, false)
        return ViewHolder(view)
    }


    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, position: Int) {
        val holder = viewHolder as ViewHolder
        val shoppingItem = dataList[position] as ShoppingItem

        holder.tv_shopping_item_ingredient.text = shoppingItem.ingredient
        holder.tv_shopping_item_amount.text = shoppingItem.amount
        holder.cb_shopping_item.isChecked = shoppingItem.isSelected
        holder.cb_shopping_item.setOnCheckedChangeListener { compoundButton, b ->
            shoppingItem.isSelected = b
        }
        holder.tv_shopping_item_unit.text = shoppingItem.unit
        holder.tv_shopping_item_category.text = shoppingItem.category

        holder.ll_shopping_item.setOnClickListener {
            onItemClickListener?.onClickListener(it,position)
        }
    }

    internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var tv_shopping_item_ingredient: TextView
        var tv_shopping_item_amount: TextView
        var cb_shopping_item: CheckBox
        var tv_shopping_item_unit: TextView
        var tv_shopping_item_category: TextView
        var ll_shopping_item:LinearLayout

        init {
            tv_shopping_item_ingredient = itemView.findViewById(R.id.tv_shopping_item_ingredient)
            tv_shopping_item_amount = itemView.findViewById(R.id.tv_shopping_item_amount)
            cb_shopping_item = itemView.findViewById(R.id.cb_shopping_item)
            tv_shopping_item_unit = itemView.findViewById(R.id.tv_shopping_item_unit)
            tv_shopping_item_category = itemView.findViewById(R.id.tv_shopping_item_category)
            ll_shopping_item = itemView.findViewById(R.id.ll_shopping_item)
        }
    }

}
